nixos/modules/config/pulseaudio.nix

```diff
@@ -52,6 +52,37 @@ let
     }
   '');
 
+  # Create a directory full of configuration files for PulseAudio to use for
+  # various modules. Packages are scanned similiar how udev does it.
+  moduleEnvVars = {
+    PA_ALSA_PATHS_DIR = "${moduleConf}/alsa-paths";
+    PA_ALSA_PROFILE_SETS_DIR = "${moduleConf}/alsa-profiles";
+  };
+  moduleConf = stdenv.mkDerivation {
+    name = "pulseaudio-moduleconf";
+
+    preferLocalBuild = true;
+    allowSubstitutes = false;
+
+    buildCommand = ''
+      mkdir -p $out/{alsa-profiles,alsa-paths}
+      shopt -s nullglob
+      set +o pipefail
+
+      function copy_dir() {
+        for j in $1/$2/*; do
+          echo "Copying $i to $out/$3/$(basename $j)"
+          cat $j > $out/$3/$(basename $j)
+        done
+      }
+
+      for i in ${toString cfg.packages}; do
+        echo "Adding configuration for package $i"
+        copy_dir $i/usr/share/pulseaudio/alsa-mixer profile-sets alsa-profiles
```

Comment:
Using `/usr` is undesirable in packages -- can this be fixed to just `/share/pulseaudio`?